### ISSUE TITLE
Browser support

### ISSUE THREAD

**Issue by felixfbecker** [kucherenko/jscpd#201]
I would like to detect clones given two strings in the browser. Is that possible?
It looks like currently the code imports a lot of Node API and modules in the same file that also contains the string-only API

**Comment by kucherenko** [comment]
Thank you for the feature request, I have plans to do that, I guess in next version I will implement the feature

**Comment by felixfbecker** [comment]
Awesome. Is there an ETA on that?
I guess all that's needed is splitting API into files that don't import Node modules

**Comment by kucherenko** [comment]
I am sorry, but currently, I fully loaded on other activities that is why I can't say clear ETA

**Comment by kucherenko** [comment]
Hi, I'm back.
I hope you still interesting in the jscpd.

Currently I'm going to release new version of jscpd@3.3.0 and test functionality with jscpd@3.3.0-rc.3.

So, jscpd moved to monorepo and part of the detector, named @jscpd/core support duplication detection in browser as well.

For use detect clones in browser use the following code:

```typescript
import {Tokenizer} from '@jscpd/tokenizer';
import {
    Detector,
    MemoryStore,
    IOptions,
    IClone,
    IStore,
    ITokenizer
} from '@jscpd/core';

const options: IOptions = {
    minLines: 5,
    maxLines: 500,
}

const tokenizer: ITokenizer = new Tokenizer();

// here you can use any store what implement IStore interface
const store: IStore = new MemoryStore();

// list of validators, implemented IValidator interface, validate clones
const validators = [];

const detector = new Detector(tokenizer, store, validators, options);

( async () => {
    const format = 'javascript';
    const code: string = '...string with code...';
    const clones: IClone[] = await detector.detect('source_id', code, format);

    console.log(clones);
})();

```
the functionality still in RC, it would be great if you can provide any feedback regarding the functionality. I will release soon to main branch and close the issue.

More info located the
[jscpd monorepo](https://github.com/kucherenko/jscpd/tree/monorepo)

**Comment by felixfbecker** [comment]
Awesome! 🎉 

So in your example, `'source_id'` would e.g. be the file path of the file and `code` would be the content? How would one tell the detector about the contents of other files to compare with? Do all the file contents have to be loaded into memory for it to work?

**Comment by kucherenko** [comment]
`source_id` is a string identifier of the code, `code` is a string with code. In general browser does not have access to file system and can't read files directly, but you can detect clones in code located on page (e.g. online IDE or textarea). During detection the code will be loaded to memory as set of tockens. I have plan add browser example soon.

**Comment by felixfbecker** [comment]
My use case would be to integrate JSCPD into a Sourcegraph plugin so it can run in the environment of the Sourcegraph webapp, on GitHub, Gitlab and other code hosts. While you are right that the browser doesn't have access to the file system, the file contents of the open repository are accessible from the API (async). But obviously loading all the contents of the repository into memory at once would not work. Would that be a requirement? Or can I provide contents for analysis file-by-file?

**Comment by kucherenko** [comment]
for the case I have plans to build REST service to detect clones, I hope it can cover your needs. Im thinking about different endpoints for detect file-by-file, difference etc.

**Comment by kucherenko** [comment]
fixed in jscpd@3.3.0-rc.9 

browser example you can find in examples folder
